content/articles/mock-a-datasource-for-react-using-json-server/index.md

```diff
@@ -0,0 +1,837 @@
+                    render={(props) => (
+                        <>
+                            {showAddBudget && <AddBudget onAdd={addBudget}/>}
+                            {budgets.length > 0 ? (
+                                <Budgets
+                                    budgets={budgets}
+                                    onDelete={deleteBudget}
+                                    onToggle={toggleReminder}
+                                />
+                            ) : (
+                                'No Budgets To Show'
+                            )}
+                        </>
+                    )}
+                />
+                <Route path='/about' component={About}/>
+                <Footer/>
+            </div>
+        </Router>
+    )
+}
+
+export default App
+```
+
+- This code shown above does the following: 
+  - Fetches the items from the specified URL and displays them
+  - Adds items to the list
+  - Deletes the items in the data source
+  - Toggles the item reminder (on, off) states
+
+### Style the application
+- Finally, do some styling to the application. The styling code that is found in the 'index.css' file:
+
+```css
+@import url('https://fonts.googleapis.com/css2?family=Poppins:wght@300;400&display=swap');
+
+:root {
+  --primary-color: #4d91d1;
+}
+
+* {
+  box-sizing: border-box;
+  margin: 0;
+  padding: 0;
+}
+
+body {
+  font-family: 'Poppins', sans-serif;
+}
+
+.container {
+  max-width: 500px;
+  margin: 30px auto;
+  overflow: auto;
+  min-height: 300px;
+  border: 1px solid steelblue;
+  padding: 30px;
+  border-radius: 5px;
+}
+
+.header {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  margin-bottom: 20px;
+  color: var(--primary-color);
+}
+
+.btn {
+  display: inline-block;
+  background: var(--primary-color);
+  color: #fff;
+  border: none;
+  padding: 10px 20px;
+  margin: 5px;
+  border-radius: 5px;
+  cursor: pointer;
+  text-decoration: none;
+  font-size: 15px;
+  font-family: inherit;
+}
+
+.btn:focus {
+  outline: none;
+}
+
+.btn:active {
+  transform: scale(0.98);
+}
+
+.btn-block {
+  display: block;
+  width: 100%;
+}
+
+.budget {
+  background: #f4f4f4;
+  margin: 5px;
+  padding: 10px 20px;
+  cursor: pointer;
+}
+
+.budget.reminder {
+  border-left: 5px solid green;
+}
+
+.budget h3 {
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+}
+
+.budget p {
+  color: #a2a2a2;
+}
+
+.add-form {
+  margin-bottom: 40px;
+}
+
+.form-control {
+  margin: 20px 0;
+}
+
+.form-control label {
+  display: block;
+}
+
+.form-control input {
+  width: 100%;
+  height: 40px;
+  margin: 5px;
+  padding: 3px 7px;
+  font-size: 17px;
+}
+
+.form-control-check {
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+}
+
+.form-control-check label {
+  flex: 1;
+}
+
+.form-control-check input {
+  flex: 2;
+  height: 20px;
+}
+
+footer {
+  margin-top: 30px;
+  text-align: center;
+}
```

Comment:
Usually, there's nothing much to explain in styling. So, this could be replaced with a GitHub/Gist link containing this CSS file.